src/category_theory/triangulated/pretriangulated.lean

```diff
@@ -0,0 +1,219 @@
+  (b : T₁.obj₂ ⟶ T₂.obj₂) (comm₁ : T₁.mor₁ ≫ b = a ≫ T₂.mor₁),
+  (∃ (c : T₁.obj₃ ⟶ T₂.obj₃), (T₁.mor₂ ≫ c = b ≫ T₂.mor₂) ∧ (T₁.mor₃ ≫ a⟦1⟧' = c ≫ T₂.mor₃) ))
+
+/--
+Given any distinguished triangle `T`, then we know `T.rotate C` is also distinguished.
+-/
+lemma rot_of_dist_triangle [CT : pretriangulated C] (T ∈ CT.distinguished_triangles) :
+  (T.rotate C ∈ CT.distinguished_triangles) :=
+by exact (pretriangulated.rotate_distinguished_triangle T).mp H
+
+/--
+Given any distinguished triangle `T`, then we know `T.inv_rotate C` is also distinguished.
+-/
+lemma inv_rot_of_dist_triangle [CT : pretriangulated C] (T ∈ CT.distinguished_triangles) :
+  (T.inv_rotate C ∈ CT.distinguished_triangles) :=
+begin
+  apply (pretriangulated.rotate_distinguished_triangle (T.inv_rotate C)).mpr,
+  exact pretriangulated.isomorphic_distinguished T H (triangulated.triangle.rotate C
+    (triangulated.triangle.inv_rotate C T)) T ((inv_rot_comp_rot C).symm.app T),
+end
+
+/--
+Given any distinguished triangle
+```
+      f       g       h
+  X  ───> Y  ───> Z  ───> X⟦1⟧
+```
+the composition `f ≫ g = 0`.
+See https://stacks.math.columbia.edu/tag/0146
+-/
+lemma comp_dist_triangle_mor_zero₁₂ [CT : pretriangulated C] (T ∈ CT.distinguished_triangles) :
+  T.mor₁ ≫ T.mor₂ = 0 :=
+begin
+  have h := pretriangulated.contractible_distinguished T.obj₁,
+  have f := CT.complete_distinguished_triangle_morphism,
+  specialize f (contractible_triangle C T.obj₁) T h H (𝟙 T.obj₁) T.mor₁,
+  have t : (triangulated.contractible_triangle C T.obj₁).mor₁ ≫ T.mor₁ = 𝟙 T.obj₁ ≫ T.mor₁,
+    by refl,
+  specialize f t,
+  cases f with c f,
+  simp at c,
+  rw ← f.left,
+  simp only [limits.zero_comp, triangulated.contractible_triangle_mor₂],
+end -- TODO : tidy this proof up
+
+/--
+Given any distinguished triangle
+```
+      f       g       h
+  X  ───> Y  ───> Z  ───> X⟦1⟧
+```
+the composition `g ≫ h = 0`.
+See https://stacks.math.columbia.edu/tag/0146
+-/
+lemma comp_dist_triangle_mor_zero₂₃  [CT : pretriangulated C] (T ∈ CT.distinguished_triangles) :
+  T.mor₂ ≫ T.mor₃ = 0 :=
+begin
+    have H₁ := rot_of_dist_triangle C T H,
+    exact comp_dist_triangle_mor_zero₁₂ C (triangulated.triangle.rotate C T) H₁
+end
+
+/--
+Given any distinguished triangle
+```
+      f       g       h
+  X  ───> Y  ───> Z  ───> X⟦1⟧
+```
+the composition `h ≫ f⟦1⟧ = 0`.
+See https://stacks.math.columbia.edu/tag/0146
+-/
+lemma comp_dist_triangle_mor_zero₃₁  [CT : pretriangulated C] (T ∈ CT.distinguished_triangles) :
+  T.mor₃ ≫ ((shift C).functor.map T.mor₁) = 0 :=
+begin
+    have H₂ := rot_of_dist_triangle C (triangle.rotate C T) (rot_of_dist_triangle C T H),
+    have t := comp_dist_triangle_mor_zero₁₂ C (triangle.rotate C (triangle.rotate C T)) H₂,
+    dsimp at t,
+    rw [comp_neg, neg_eq_zero] at t,
+    exact t,
+end
+
+/-
+TODO: If `C` is pretriangulated with respect to a shift,
+then `C^{op}` is pretriangulated with respect to the inverse shift.
```

Comment:
I don't know what is standard in mathlib, but you could consider writing C^{op} as C\^o\^p since it is a comment anyway?